### DISCUSSION TITLE
Client Connection Issue After Redis Cluster Restart in Kubernetes

### DISCUSSION THREAD

**Discussion by captainpro-eng** [OT-CONTAINER-KIT/redis-operator#1081] [Q&A]
I am experiencing an issue with my Redis cluster deployed in Kubernetes. When the Redis cluster is restarted, the client encounters the following exception:

Could not get a resource from the pool; nested exception is redis.clients.jedis.exceptions.JedisConnectionException: Could not get a resource from the pool.

Problem:

After the Redis cluster restarts, the client attempts to connect using the old pod IP addresses, which are no longer valid since the pods have received new IP addresses. This leads to connection failures and the inability to interact with the Redis cluster.

Steps to Reproduce:

Deploy a Redis cluster in a Kubernetes environment.
Connect a client (using Jedis or another library) to the Redis cluster.
Restart the Redis cluster (e.g., by scaling down and back up).
Observe that the client throws the connection exception.
Expected Behavior:

After the Redis cluster restarts, the client should automatically reconnect to the new pod IPs or hostnames, avoiding any connection exceptions.

**Comment by captainpro-eng** [comment]
++ @drivebyer

**Comment by shubham-cmyk** [comment]
@captainpro-eng is your issue related to this dicussion : https://github.com/OT-CONTAINER-KIT/redis-operator/discussions/1080

**Comment by drivebyer** [comment]
@captainpro-eng 
We usually connect to the Redis cluster using the FQDN of the Redis pod name or the service on top of the Redis pod. This way, even after a pod restart, we can still connect to the correct pod through Kubernetes DNS resolution.

**Reply by captainpro-eng** [reply]
@drivebyer  The Spring Data Redis client is not connecting to the new pod. I’m using a service in the pool configuration. After the pool is created, the client should fetch all pod IPs and connect to all of them. However, I restarted Redis, and the client is still trying to connect to the old IP, resulting in an exception: "can't get resource pool."
